backend/packages/Upgrade/src/api/repositories/AnalyticsRepository.ts

```diff
@@ -418,100 +411,110 @@ export class AnalyticsRepository {
-      .addGroupBy('"enrollingStateTimeLog"."timeLog"')
-      .addGroupBy('"enrollmentCompleteStateTimeLog"."timeLog"')
-      .addGroupBy('"monitoredPointLogs"."createdAt"')
-      .addGroupBy('"individualEnrollment"."enrollmentCode"')
-      .addGroupBy('"individualExclusion"."exclusionCode"')
-      .addGroupBy('"userStratificationFactor"."stratificationFactorValue"')
-      .addGroupBy('"monitoredDecisionPoint"."userId"')
-      .orderBy('"monitoredDecisionPoint"."userId"', 'ASC')
+      .orderBy('monitoredDecisionPoint.userId', 'ASC')
       .offset(skip)
       .limit(take)
-      .where('"monitoredDecisionPoint"."experimentId" = :experimentId', { experimentId })
-      .execute();
+      .where('monitoredDecisionPoint.experimentId = :experimentId', { experimentId });
+    
+    const [individualEnrollmentQueryResults, monitoredDecisionPointQueryResults] = await Promise.all([
+      individualEnrollmentQuery.getRawMany(),
+      monitoredDecisionPointQuery.getRawMany(),
+    ]);
+
+    const userStratificationFactorUserList = [];
+
+    const individualEnrollmentExperimentData = [];
+    individualEnrollmentQueryResults.forEach(individualEnrollmentQueryResult => {
+      experimentsData.forEach((experiment) => {
+        if (experiment.expDecisionPointId === individualEnrollmentQueryResult.partitionId &&
+            experiment.expConditionId === individualEnrollmentQueryResult.conditionId) {
+          // prepare users list that have SRS experiment:
+          if (experiment.stratification) {
+            userStratificationFactorUserList.push(individualEnrollmentQueryResult.userId)
+          }
+          individualEnrollmentExperimentData.push({
+            ...experiment,
+            userId: individualEnrollmentQueryResult.userId,
+            groupId: individualEnrollmentQueryResult.enrollmentGroupId,
+            enrollmentCode: individualEnrollmentQueryResult.enrollmentCode,
+            stratification: experiment.stratification,
+          });
+        }
+      });
+    });
+
+    let userStratificationFactorQueryResult = [];
+    if (isSRSExperiment) {
+      // get users stratification factor values:
+      const userStratificationFactorRepository = getCustomRepository(UserStratificationFactorRepository, 'export');
+
+      const userStratificationFactorQuery = userStratificationFactorRepository
+        .createQueryBuilder('userStratificationFactor')
+        .select([
+          'userStratificationFactor.user as "userId"',
+          'userStratificationFactor.stratificationFactorValue as "stratificationFactorValue"'
+        ])
+        .where('userStratificationFactor.user IN (:...userIds)', { userIds: userStratificationFactorUserList.map((data) => data) });
+      
+      userStratificationFactorQueryResult = await userStratificationFactorQuery.getRawMany();
+    }
+    // Combined results
+    const combinedCSVExportData = [];
+
+    individualEnrollmentExperimentData.forEach(individualEnrollmentExperiment => {
```

Comment:
It would be best if you used `lodash` groupBy or something similar on the `monitoredDecisionPointQuery` result to groupBy userId to avoid nested loops.